runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java

```diff
@@ -203,13 +207,22 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 			}
 			else
 			{
+				// Prefer the world-based endpoint if the world is special
+				endpoint = getWorldEndpoint();
+				if (endpoint == HiscoreEndpoint.NORMAL)
```

Comment:
Players can dm you from other worlds, so it probably would be best to just not check this. Or maybe check the flags of the world they are in, but those aren't always available.